ray-curator/ray_curator/examples/wikipedia_example.py

```diff
@@ -0,0 +1,168 @@
+    if hasattr(task, "data") and task.data is not None:
+        if hasattr(task.data, "head"):  # pandas DataFrame
+            print("  Sample data:")
+            print(task.data.head(3))
+        elif isinstance(task.data, list) and len(task.data) > 0:
+            print(f"  Sample items ({len(task.data)} total):")
+            for j, item in enumerate(task.data[:3]):
+                print(f"    Item {j + 1}: {item}")
+        else:
+            print(f"  Data: {task.data}")
+
+
+def display_results(results: list, output_dir: str) -> None:
+    """Display results from the pipeline execution."""
+    if results:
+        print("\nResults from pipeline:")
+        for i, task in enumerate(results):
+            display_task_info(task, i)
+            if i >= MAX_DISPLAY_TASKS - 1:  # Show only first 3 tasks
+                break
+    else:
+        print("No results returned from pipeline")
+
+    # Check if JSONL files were created
+    if os.path.exists(output_dir):
+        jsonl_files = [f for f in os.listdir(output_dir) if f.endswith(".jsonl")]
+        if jsonl_files:
+            print(f"\nGenerated JSONL files: {jsonl_files}")
+        else:
+            print("\nNo JSONL files were generated")
+
+
+def main() -> None:
+    """Run the Wikipedia download and extract example."""
+    # Create directories
+    temp_dir, download_dir, output_dir = create_directories()
+
+    print(f"Temporary directory: {temp_dir}")
+    print(f"Download directory: {download_dir}")
+    print(f"Output directory: {output_dir}")
+
+    try:
+        # Create pipeline
+        pipeline, wikipedia_stage = create_pipeline(download_dir, output_dir)
+
+        print("\nStarting Wikipedia download for language: en")
+        print(f"Stage description: {wikipedia_stage.get_description()}")
+
+        # Print pipeline description
+        print("\nPipeline description:")
+        print(pipeline.describe())
+        print("\n" + "=" * 50 + "\n")
+
+        # Run the pipeline
+        executor = XennaExecutor()
+        print("Starting pipeline execution...")
+        results = pipeline.run(executor)
+
+        print("\nPipeline completed successfully!")
+        print(f"Total output tasks: {len(results) if results else 0}")
+        print(f"Results saved to: {output_dir}")
+
+        # Display results
+        display_results(results, output_dir)
+
+    except Exception as e:
+        print(f"Error running pipeline: {e}")
+        raise
+
+    finally:
+        # Clean up temporary directory
+        print(f"\nCleaning up temporary directory: {temp_dir}")
+
+
+if __name__ == "__main__":
+    import time
+
+    start_time = time.time()
+    main()
+    end_time = time.time()
+    print(f"Total execution time: {end_time - start_time} seconds")
+    # Execution time for Xenna: 97s
+    # Execution time for RayData: 96s
+    # Both produce identical results (md5sums match)
+    # 199 records in total across 2 output files.
```

Comment:
Can you add time taken for each stage? I'm curious what the breakdown is? Also add the numbers here https://docs.google.com/document/d/1ULUdvHQhlplDEME3gU_3nKXREDjyzoMTGTuG_KkNlbU/edit?tab=t.0

Reply:
Can you do a check with dask too and see if results match with respect to columns?